tests/IntegrationTests/WebAppUiTests/UiTestHelpers.cs

```diff
@@ -1,70 +1,275 @@
+        }
+        private static void WriteLine(ITestOutputHelper? output, string message)
+        {
+            if (output != null)
+            {
+                output.WriteLine(message);
+            }
+            else
+            {
+                Trace.WriteLine(message);
+            }
+        }
+
+        /// <summary>
+        /// This starts the recording of playwright trace files. The corresponsing EndAndWritePlaywrightTrace method will also need to be used.
+        /// This is not used anywhere by default and will need to be added to the code if desired
+        /// </summary>
+        /// <param name="page">The page object whose context the trace will record</param>
+        /// <returns></returns>
+        public static async Task StartPlaywrightTrace(IPage page) 
+        {
+            await page.Context.Tracing.StartAsync(new()
+            {
+                Screenshots = true,
+                Snapshots = true,
+                Sources = true
+            });
+        }
+
+        /// <summary>
+        /// This file gets written to the test's bin/debug/[dotnet version] folder
+        /// Use the app at https://trace.playwright.dev/ to easily view the trace, all trace data is opened locally by the web app.
+        /// </summary>
+        /// <param name="page">The page object whose context is recording a trace</param>
+        /// <returns>Nothing just creates the file</returns>
+        public static async Task EndAndWritePlaywrightTrace(IPage page) 
+        {
+            await page.Context.Tracing.StopAsync(new()
+            {
+                Path = "PlaywrightTrace.zip"
+            });
+        }
+
+        /// <summary>
+        /// Starts a process from an executable, sets its working directory, and redirects its output to the test's output
+        /// </summary>
+        /// <param name="testAssemblyLocation">The path to the test's directory</param>
+        /// <param name="appLocation">The path to the processes directory</param>
+        /// <param name="executableName">The name of the executable that launches the process</param>
+        /// <param name="portNumber">The port for the process to listen on</param>
+        /// <param name="isHttp">If the launch URL is http or https. Default is https.</param>
+        /// <returns></returns>
+        public static Process? StartProcessLocally(string testAssemblyLocation, string appLocation, string executableName, uint? portNumber = null, bool? isHttp = false)
+        {
+            string applicationWorkingDirectory = GetApplicationWorkingDirectory(testAssemblyLocation, appLocation);
+            ProcessStartInfo processStartInfo = new ProcessStartInfo(applicationWorkingDirectory + executableName);
+            processStartInfo.WorkingDirectory = applicationWorkingDirectory;
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.RedirectStandardError = true;
+
+            if (portNumber != null && portNumber > 0)

```

Comment:
here and below, it is preferable to use `.HasValue` instead of a null check for nullables

Reply:
done